File: AtCoder/abc/abc254/d.cpp
#include <bits/stdc++.h>
#define rep(i, n) for(int i = 0; i < (n); ++i)
#define rep1(i, n) for(int i = 1; i < (n); ++i)
using namespace std;
using ll = long long;

ll n;

ll count(ll num){
    ll sq = num*num;
    if(sq == 1) return 1;

    ll ret = 1;
    for(ll i = num-1; i >= 1; --i){
        if(sq%i == 0 && sq/i <= n) ret += 2;
        if(sq/i > n) break;
    }
    
    return ret;
}

int main(){
    cin.tie(nullptr);
    ios::sync_with_stdio(false);

    // cout << count(4) << " " << count(9) << " " << count(16) << endl;
    cin >> n;

    ll res = 0;
    for(ll i = 1; i <= n; ++i){
        res += count(i);

        // cout << i << " " << sq << " " << count(sq) << endl;

        // if(i*i > n) break;
    }

    cout << res << endl;

    return 0;
}

File: AtCoder/abc/abc254/d.py
from collections import defaultdict

def calc(num):

    pf = defaultdict(int)
    for i in range(2, num):
        if i*i > num: break

        while num % i == 0:
            pf[i] += 1
            num //= i
        
    if num != 1:
        pf[num] = 1


    ret = 1
    for k in pf:
        if pf[k] % 2 == 1:
            ret *= k

    return ret



def main():

    n = int(input())    

    res = 0
    for i in range(1, n+1):
        k = calc(i)

        nk = k
        res += 1
        for j in range(2, n+1):
            nk = k * j*j

            if nk <= n:
                res += 1
            else:
                break
    
    print(res)






if __name__ == '__main__':
    main()